src/transfers/MultiAppMooseOkapiTransfer.C

```diff
@@ -69,79 +89,59 @@ MultiAppMooseOkapiTransfer::execute()
-                  "MultiAppMooseOkapiTransfer FROM_MULTIAPP "
-                  "are not all the same!");
+          std::vector<Real> & coefficients = from_object.getCoefficients();
 
           // Initialize the moose_coeffs array and then populate by looping over
           // the source variables.
-          int num_coeffs_from_moose = num_vars_to_read * source_var_size;
-          double moose_coeffs[num_coeffs_from_moose];
-          for (auto i = beginIndex(_src_var_names); i < num_vars_to_read; ++i)
-          {
-            auto & soln_values = source_vars[i]->sln();
-            for (auto j = beginIndex(soln_values); j < soln_values.size(); ++j)
-              moose_coeffs[i * num_vars_to_read + j] = soln_values[j];
-          }
+          size_t num_coeffs_from_moose = coefficients.size();
+          double * moose_coeffs = &coefficients[0];
 
           if (_dbg)
           {
-            _console << "Transferring " << num_coeffs_from_moose <<
-              " coefficients from MOOSE to OpenMC for cell " << _cell << std::endl;
+            _console << "Transferring " << num_coeffs_from_moose
+                     << " coefficients from MOOSE to OpenMC for cell " << _cell << std::endl;
 
             for (int i = 0; i < num_coeffs_from_moose; ++i)
               _console << moose_coeffs[i] << " ";
             _console << std::endl;
           }
 
           // send all expansion coefficients for a cell to OpenMC
-          int err_recv  = OpenMC::receive_coeffs_for_cell(_index,
-            moose_coeffs, num_coeffs_from_moose);
+          int err_recv =
+              OpenMC::receive_coeffs_for_cell(_index, moose_coeffs, num_coeffs_from_moose);
           ErrorHandling::receive_coeffs_for_cell(err_recv);
 
-        // Change a temperature in OpenMC. For now, only use a single coefficient,
-        // since there's no continuous material tracking yet.
-        Real temp = (source_vars[0]->sln())[0] / sqrt(2.0 * M_PI);
-        if (_dbg) _console << "Setting OpenMC cell " << _cell <<
-          " temperature to " << temp << std::endl;
+          // Change a temperature in OpenMC. For now, only use a single coefficient,
+          // since there's no continuous material tracking yet.
+          Real temp = moose_coeffs[0] / sqrt(2.0 * M_PI);
+          if (_dbg)
+            _console << "Setting OpenMC cell " << _cell << " temperature to " << temp << std::endl;
 
-        if (_store_results)
-        {
-          std::vector<Real> this_iteration;
-          for (int i = 0; i < num_coeffs_from_moose; ++i)
-            this_iteration.push_back(moose_coeffs[i]);
+          if (_store_results)
+          {
+            std::vector<Real> this_iteration;
+            for (int i = 0; i < num_coeffs_from_moose; ++i)
+              this_iteration.push_back(moose_coeffs[i]);
 
-          _fuel_temp_coeffs.push_back(this_iteration);
+            _fuel_temp_coeffs.push_back(this_iteration);
```

Comment:
I think we can get rid of `_store_results` if the FE coefficients can be saved in an output file by MOOSE; maybe this can already be done, or we could add something to the FE module (can be done in a future PR)?

Reply:
I don't believe we currently have a method in the FX module for outputting coefficients, but it would definitely be straightforward to add. Should we keep this for now until that method exists?

Reply:
Yeah, I think this is fine for now.